### PR TITLE
[helm] Add pod annotations, labels, and pod disruption budget support

### LINKED ISSUE #3115: [helm] Add pod annotations, pod labels, and PodDisruptionBudget support
### Search before asking

- [x] I searched in the [issues](https://github.com/apache/fluss/issues) and found nothing similar.

### Description

The Fluss Helm chart is missing standard Kubernetes pod-level configuration that production deployments rely on. 

There is currently no way to add custom annotations to pods (needed for Prometheus scraping, Vault agent injection, Datadog, etc.), no way to add custom labels (needed for service mesh routing, monitoring selectors), and no PodDisruptionBudget to protect against draining too many replicas during rolling upgrades or node maintenance. 

### Willingness to contribute

- [x] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by fresh-borzoni**
closes https://github.com/apache/fluss/issues/3115

Production Kubernetes deployments commonly need custom pod annotations for integrations like Prometheus scraping or Vault agent injection, custom labels for service mesh routing and monitoring selectors, and PodDisruptionBudgets to prevent simultaneous eviction of tablet server replicas during node maintenance. 

This PR adds all three as standard Helm chart fields for both coordinator and tablet server components, with PDB disabled by default.

### CODE REVIEW

**Comment by fresh-borzoni** [comment]
@affo @morazow @polyzos PTAL 🙏

File: helm/templates/pdb-coordinator.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+metadata:
+  name: coordinator-server
+  labels:
+  {{- include "fluss.labels" . | nindent 4 }}
+spec:
+  {{- with .Values.coordinator.podDisruptionBudget.minAvailable }}
+  minAvailable: {{ . }}
+  {{- end }}
+  {{- with .Values.coordinator.podDisruptionBudget.maxUnavailable }}
+  maxUnavailable: {{ . }}
```
Comment (morazow) [inline helm/templates/pdb-coordinator.yaml]: Reading the docs on the PDB: https://kubernetes.io/docs/tasks/run-application/configure-pdb/#specifying-a-poddisruptionbudget

One can only specify either `minAvailable` or `maxUnavailable`. Could you please add validation for this?

You can check the `NOTES.txt` for example for now, until https://github.com/apache/fluss/pull/2902 is merged 🤝

Comment (morazow) [reply helm/templates/pdb-coordinator.yaml]: Valid also for tablet PDB

Comment (morazow) [inline helm/templates/pdb-coordinator.yaml]: On similar note, I could not find what will happen if user doesn't set any of `minAvailable` or `maxAvailable`, seems undefined.

To be on safe side, let's also add a validation that one of them should be set when PDB is enabled.

File: helm/tests/pod_metadata_and_pdb_test.yaml [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+suite: coordinator-pdb
+templates:
+  - templates/pdb-coordinator.yaml
+
+tests:
+  - it: should render coordinator PDB with minAvailable
+    set:
+      coordinator.podDisruptionBudget:
+        enabled: true
+        minAvailable: 1
```
Comment (morazow) [inline helm/tests/pod_metadata_and_pdb_test.yaml:200]: Let's also add test with `minAvailable` and `maxUnavailable` set to zero 🤝

File: helm/templates/sts-coordinator.yaml [hunk 3]
```diff
@@ -34,6 +34,13 @@ spec:
       labels:
         {{- include "fluss.selectorLabels" . | nindent 8 }}
         app.kubernetes.io/component: coordinator
+        {{- with .Values.coordinator.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
```
Comment (morazow) [inline helm/templates/sts-coordinator.yaml:37]: Should we put the `podLabels` before the `selectorLabels`?

I wonder what will happen if the `podLabels` overwrite the `selectorLabels`, definitely the matchLabels would be different.

In any case, either again we could validate that podLabels does not contain any selectorLabels, or change the ordering so that selectorLabels overwrite previously set ones (if any).

**Review by morazow — CHANGES_REQUESTED** [review]
Thanks man @fresh-borzoni 🚀 

I have added some comments and requests 🤝

File: helm/templates/sts-coordinator.yaml (hunk 3 above)
Comment (fresh-borzoni) [reply helm/templates/sts-coordinator.yaml:37]: Reordered, good point 👍

**Comment by fresh-borzoni** [comment]
@morazow Ty for the nice review, addressed, PTAL 🙏

**Review by morazow — APPROVED** [review]
Looks good 🤝

**Review by affo — APPROVED** [review]
Another fundamental one, thanks @fresh-borzoni 🚀
